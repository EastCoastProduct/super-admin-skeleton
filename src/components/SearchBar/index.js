import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './index.style';

const SearchBar = ({ style }) =>
  <div className={css(styles.bar, style && style)}>
    <i className={`${css(styles.icon)} fa fa-search`} />
    <input className={css(styles.input)} />
  </div>;

SearchBar.propTypes = {
  style: PropTypes.object.isRequired,
};

export default SearchBar;
