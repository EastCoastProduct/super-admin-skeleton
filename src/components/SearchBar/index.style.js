import { StyleSheet } from 'aphrodite/no-important';
import variables from '../../styles/variables';

const { colors, sizes } = variables;

export default StyleSheet.create({
  bar: {
    background: colors.input,
    borderRadius: sizes.radiusBig,
    boxShadow: `inset 1px 1px ${colors.shadowAlt}`,
    padding: '8px 15px',
  },
  icon: {
    display: 'inline-block',
    fontSize: 24,
    verticalAlign: 'middle',
    width: 24,
  },
  input: {
    border: 0,
    color: colors.font,
    display: 'inline-block',
    outline: 0,
    padding: '0 10px',
    verticalAlign: 'middle',
    width: 'calc(100% - 24px)',
  },
});
