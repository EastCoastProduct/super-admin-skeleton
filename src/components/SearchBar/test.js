import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheet } from 'aphrodite/no-important';
import SearchBar from './';

describe('SearchBar component snapshot', () => {
  it('renders basic required data', () => {
    const wrapper = shallow(<SearchBar input={{}} meta={{}} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with additional style, value and extra attributes', () => {
    const styles = StyleSheet.create({
      additional: {
        margin: 10,
        padding: 10,
      },
    });
    const wrapper = shallow(
      <SearchBar
        className={styles.additional}
        input={{ value: 'name' }}
        meta={{}}
        placeholder="This is Search Box"
        type="text"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
