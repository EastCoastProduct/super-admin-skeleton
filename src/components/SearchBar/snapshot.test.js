import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheet, StyleSheetTestUtils } from 'aphrodite/no-important';
import SearchBar from './';

describe('SearchBar component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data', () => {
    const tree = renderer.create(<SearchBar input={{}} meta={{}} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders with additional style and value', () => {
    const style = StyleSheet.create({
      searchBar: {
        margin: 10,
        padding: 10,
      },
    });
    const tree = renderer.create(
      <SearchBar style={style.searchBar} input={{ value: 'name' }} meta={{}} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
