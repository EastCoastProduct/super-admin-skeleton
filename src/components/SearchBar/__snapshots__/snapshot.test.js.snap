exports[`SearchBar component snapshot renders basic required data 1`] = `
<div
  className="bar_t04vgp">
  <button
    className="empty_1b6lvt9"
    type="submit">
    <i
      className="icon_48j6am fa fa-search" />
  </button>
  <input
    className="input_bvvtu9" />
</div>
`;

exports[`SearchBar component snapshot renders with additional style, value and extra attributes 1`] = `
<div
  className="bar_t04vgp-o_O-searchBar_70d19d">
  <button
    className="empty_1b6lvt9"
    type="submit">
    <i
      className="icon_48j6am fa fa-search" />
  </button>
  <input
    className="input_bvvtu9"
    placeholder="This is Search Box"
    value="name" />
</div>
`;
